app/controllers/changes_controller.rb

```diff
@@ -23,6 +23,34 @@ def resend
     render json: @change.send_notifications
   end
 
+  def html
+    @change_id=params[:change_id]
+    @snapshot = PageSnapshot.find(@change_id)
+    @page = Page.find(@snapshot.page_id)
+    @timestamp = @snapshot.created_at
+    @html = @snapshot.html
+  end
+
+  def raw_html
+    @change_id=params[:change_id]
+    @snapshot = PageSnapshot.find(@change_id)
+    return render html: @snapshot.html.html_safe
+  end
+
+
+  def download
+    @id = params[:change_id]
+    @change_id = @id
+    @snapshot = PageSnapshot.find(@id)
+    @html = @snapshot.html
+    @parent_page = Page.find(@snapshot.page_id)
+    @filename = @parent_page.name.gsub(" ","-") + "-" + @snapshot.created_at.to_s.gsub(" ","-") + ".html"
+    send_data(@html,
+          :filename => @filename,
+          :type => 'text/html',
+          :disposition => 'attachment') 
+  end
```

Comment:
Basically the same notes as the html method, but one additional thing is that you don't need to make these instance variables (prepending the `@`), since you're just using them locally in this method. You might also consider moving the filename logic to the PageSnapshot model. Feels like a more natural home for it.